lib/ansible/module_utils/urls.py

```diff
@@ -755,6 +762,35 @@ def extract_pem_certs(b_data):
         yield match.group(0)
 
 
+class GzipDecodedReader(gzip.GzipFile if HAS_GZIP else object):
+    """A file-like object to decode a response encoded with the gzip
+    method, as described in RFC 1952.
+
+    Largely copied from ``xmlrpclib``/``xmlrpc.client``
+    """
+    def __init__(self, fp):
+        if not HAS_GZIP:
+            raise NotImplementedError
```

Comment:
would it not be better to also mention that it requires gzip python lib to work?

Reply:
Yes, you are correct. I assumed I had that friendly message reported somewhere, and maybe I did at one point, but it's missing now.

Reply:
Fixed.

Reply:
```
localhost | FAILED! => {
    "changed": false,
    "content": "",
    "elapsed": 0,
    "msg": "Status code was -1 and not [200]: An unknown error occurred: Python gzip support required to decompress gzip encoded responses",
    "redirected": false,
    "status": -1,
    "url": "http://httpbin.org/gzip"
}
```